File: ocaml/04-control-flow/conditionals.ml
(* if/else is an EXPRESSION in OCaml - it evaluates to a value *)

let () =
  let temperature = 18 in

  (* Used for side effects: both branches return unit *)
  if temperature > 25 then
    print_endline "It's warm outside"
  else
    print_endline "Bring a jacket";

  (* Used as an expression: the result is bound to a name *)
  let label = if temperature > 20 then "warm" else "cool" in
  Printf.printf "The weather is %s\n" label;

  (* Chained conditions with else if *)
  let category =
    if temperature < 0 then "freezing"
    else if temperature < 15 then "cold"
    else if temperature < 25 then "mild"
    else "hot"
  in
  Printf.printf "Category: %s\n" category;

  (* An if with no else returns unit, so the branch must be unit too *)
  if temperature < 20 then print_endline "Below 20 degrees"

File: scala/04-control-flow/Conditionals.scala
object Conditionals {
  def main(args: Array[String]): Unit = {
    val temperature = 18

    // if/else used as a statement for side effects
    if (temperature > 25) {
      println("It's warm outside")
    } else if (temperature > 10) {
      println("It's mild outside")
    } else {
      println("It's cold outside")
    }

    // if/else used as an EXPRESSION — it returns a value
    val label = if (temperature > 20) "warm" else "cool"
    println(s"The weather is $label")

    // Scala has no ternary operator; if/else fills that role
    val adjusted = if (temperature > 20) temperature else 20
    println(s"Adjusted reading: $adjusted")
  }
}
